zerver/lib/actions.py

```diff
@@ -309,20 +309,29 @@ def add_new_user_history(user_profile, streams):
 # * Notifies other users in realm and Zulip about the signup
 # * Deactivates PreregistrationUser objects
 # * subscribe the user to newsletter if newsletter_data is specified
-def process_new_human_user(user_profile, prereg_user=None, newsletter_data=None):
-    # type: (UserProfile, Optional[PreregistrationUser], Optional[Dict[str, str]]) -> None
+def process_new_human_user(user_profile, prereg_user=None, newsletter_data=None, default_stream_group_names=[]):
+    # type: (UserProfile, Optional[PreregistrationUser], Optional[Dict[str, str]], List[str]) -> None
     mit_beta_user = user_profile.realm.is_zephyr_mirror_realm
     if prereg_user is not None:
-        streams = prereg_user.streams.all()
+        streams = set(prereg_user.streams.all())
```

Comment:
I'm not sure this is a great idea; I seem to recall that Django can do unexpected things when working with sets of model objects...